packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,554 @@
+
+# Overview
+
+Flow Client Library (FCL) approaches the idea of blockchain wallets on Flow in a different way than how wallets may be supported on other blockchains. For example, with FCL, a wallet is not necessarily limited to being a browser extention or even a native application on a users device. FCL offers wallet developers the flexibility and freedom to build many different types of applications. Since wallet applications can take on many forms, we needed to create a way for these varying applications to be able to communicate and work together.
+
+FCL acts in many ways as a protocol to facilitate communication and configuration between the different parties involved in a blockchain application. An _Application_ can use FCL to _authenticate_ users, and request _authorizations_ for transactions, as well as mutate and query the _Blockchain_. An application using FCL offers it's _Users_ a way to connect and select any number of Wallets Providers and their Wallet Services. A selected _Wallet_ provides an Applications instance of FCL with configuration information about itself and its Wallet Services, allowing the _User_ and _Application_ to interact with them.
+
+In the following paragraphs we'll explore ways in which you, as a wallet developer, can integrate with FCL by providing implementataions of various FCL services. 
+
+The following services will be covered:
+- Authentication (Authn) Service
+- Authorization (Authz) Service
+- User Signature Service
+- Pre-Authz Service
+
+# Service Methods
+
+FCL Services are your way as a Wallet Provider of configuring FCL. In accomplisshing these service, FCL employs various Service Methods. You can think of these Service Methods as the strategies FCL and your Wallet Provider will use to provide the functionality of your wallets supported FCL services.
+
+Sometimes services just configure FCL and thats it. An example of this case can be seen with the Authentication Service and the OpenID Service.
+With those two services you are simply telling FCL "here is a bunch of info about the current user". (You will see that those two services both have a `method: "DATA"` field in them.
+Currently these are the only two cases that can be a data service.)
+
+Other services can be a little more complex. For example, they might require a back and forth communication between FCL and the Service in question.
+Ultimately we want to do this back and forth via a secure back-channel (https requests to servers), **but in some situations that isn't a viable option, so there is also a front-channel option**.
+Where possible, you should aim to provide a back-channel support for services, and only fall back to a front-channel if absolutely necessary.
+
+Back-channel communications use `method: "HTTP/POST"`, while front-channel communications use `method: "IFRAME/RPC"`, `method: "POP/RPC"` or `method: "TAB/RPC`.
```

Comment:
This is probably a good place for a table:
```
       Wat | Front |  Back
-----------+-------+------
 HTTP/POST |   -   |   x
IFRAME/RPC |   x   |   - 
   POP/RPC |   x   |   -
   TAB/RPC |   x   |   -
```

Reply:
Added this table